### PR TITLE
Update QD to reflect QL 2 statuses.

### PR DESCRIPTION
**Pull request by brawner**
Updates rosidl_runtime_cpp and rosidl_typesupport_interface.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: rosidl_runtime_cpp/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -99,7 +99,17 @@ The BoundedVector class is tested and the most recent test results can be found
...
+
+This includes:
+
+- tracking and reporting line coverage statistics
+- achieving and maintaining a reasonable branch line coverage (90-100%)
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed [here](https://ci.ros2.org/job/ci_linux_coverage/lastBuild/cobertura/src_rosidl_rosidl_runtime_cpp_include_rosidl_runtime_cpp/).
```
Comment (ahcorde) [inline rosidl_runtime_cpp/QUALITY_DECLARATION.md:112]: The link is not wroking for me: https://ci.ros2.org/job/ci_linux_coverage/lastBuild/cobertura/src_ros2_rosidl_rosidl_runtime_cpp_include_rosidl_runtime_cpp/

Comment (brawner) [reply rosidl_runtime_cpp/QUALITY_DECLARATION.md]: Updated it to lastSuccessfulBuild.

File: rosidl_runtime_cpp/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -99,7 +99,17 @@ The BoundedVector class is tested and the most recent test results can be found
...
+
+This includes:
+
+- tracking and reporting line coverage statistics
+- achieving and maintaining a reasonable branch line coverage (90-100%)
+- no lines are manually skipped in coverage calculations
+
+Changes are required to make a best effort to keep or increase coverage before being accepted, but decreases are allowed if properly justified and accepted by reviewers.
+
+Current coverage statistics can be viewed [here](https://ci.ros2.org/job/ci_linux_coverage/lastSuccessfulBuild/cobertura/src_rosidl_rosidl_runtime_cpp_include_rosidl_runtime_cpp/).
```
Comment (Blast545) [inline rosidl_runtime_cpp/QUALITY_DECLARATION.md:112]:
```suggestion
Current coverage statistics can be viewed [here](https://ci.ros2.org/job/ci_linux_coverage/lastSuccessfulBuild/cobertura/src_ros2_rosidl_rosidl_runtime_cpp_include_rosidl_runtime_cpp/).
```

File: rosidl_runtime_cpp/QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -99,7 +99,18 @@ The BoundedVector class is tested and the most recent test results can be found
...
+A summary of how these statistics are calculated can be found in the ["ROS 2 On-boarding guide"](https://index.ros.org/doc/ros2/Contributing/ROS-2-On-boarding-Guide/#note-on-coverage-runs).
```
Comment (hidmic) [inline rosidl_runtime_cpp/QUALITY_DECLARATION.md:113]: @brawner consider removing the quotes around `ROS 2 On-boarding guide`.

Comment (brawner) [reply rosidl_runtime_cpp/QUALITY_DECLARATION.md]: Done

File: rosidl_runtime_cpp/QUALITY_DECLARATION.md (hunk 2 above)
Comment (brawner) [reply rosidl_runtime_cpp/QUALITY_DECLARATION.md]: Fixed.

**Pull request merged**